compiler/rustc_codegen_llvm/src/gotoc/statement.rs

```diff
@@ -106,22 +106,35 @@ impl<'tcx> GotocCtx<'tcx> {
             let le = self.codegen_place(location).goto_expr;
             hk.handle(self, instance, vec![le], None, Some(*target), None)
         } else {
-            let drop_implementation = match instance.def {
-                InstanceDef::Virtual(..) => self
-                    .codegen_unimplemented(
-                        "drop_in_place for dynamic argument",
-                        Type::Empty,
-                        Location::none(),
-                        "https://github.com/model-checking/rmc/issues/11",
-                    )
-                    .as_stmt(Location::none()),
-                InstanceDef::DropGlue(..) => Stmt::skip(Location::none()),
-                _ => {
-                    let func = self.codegen_func_expr(instance, None);
-                    func.call(vec![self.codegen_place(location).goto_expr.address_of()])
-                        .as_stmt(Location::none())
+            // Extract nested object if type is pointed to by a fat pointer
+            let nested_ty = match loc_ty.kind() {
+                ty::Ref(_, pointee_type, ..) => pointee_type,
+                ty::Adt(adt_def, adt_subst) => {
+                    if adt_def.is_box() || self.vtable_field_name(adt_def.did).contains("cell") {
```

Comment:
Is there a helper function somewhere that already does this? What does the llvm code do here?

Reply:
The LLVM code does not handle boxes and cells specially. I am rethinking this approach.